### ISSUE TITLE
Project Broken Due To Docker / Docker Compose - Unsupported Compose File Version - Root Additional Property Name Is Not Allowed

### ISSUE THREAD

**Issue by akamspsw** [tiangolo/full-stack-fastapi-postgresql#486]
This project does not work anymore.

With a clean cookiecutter, if you follow the directions in dockerswarm.rocks you get the error: `"Unsupported compose file version 1.0"`

This happens when you get to `docker stack` in /scripts/deploy.sh

_I think it's because docker stack and docker compose are not compatible._

If you try to add a version to the top of the file after the compose config, i.e.  ...

```
(echo -e "version: '3.8'\n"; DOMAIN=${DOMAIN?Variable not set} \
TRAEFIK_TAG=${TRAEFIK_TAG?Variable not set} \
STACK_NAME=${STACK_NAME?Variable not set} \
TAG=${TAG?Variable not set} \
docker-compose \
-f docker-compose.yml \
config)
```

... then you still get this error `"(root) additional property **name** is not allowed"`

**Comment by akamspsw** [comment]
This is because tiangolo updated docker-with-compose image.

update .gitlab-ci.yml will fix....

`tiangolo/docker-with-compose:2021-09-18`

**Comment by tiangolo** [comment]
Yep, there's a lot of things I have to update here, starting with migrating to SQLModel. But first I have to finish dinner things in SQLModel. 🤓

**Comment by lacriment** [comment]
For quick solution update your `.gitlab-ci.yml` file with following line:

```
image: tiangolo/docker-with-compose:2021-09-18
```

I do not like breaking changes.

**Comment by SobieskiCodes** [comment]
> Yep, there's a lot of things I have to update here, starting with migrating to SQLModel. But first I have to finish dinner things in SQLModel. 🤓

Glad to hear there are plans on the board to update this, I appreciate the project - thank you. I've been working on a project structure loosely based on your project for the last month or so. Do you have any major adjustments for the front end in mind at this stage, or is that loosely going to stay intact?

**Comment by YuriiMotov** [comment]
Resolved
